Lib/pdb.py

```diff
@@ -2077,34 +2075,44 @@ def help():
 
 
 def main():
-    import getopt
-
-    opts, args = getopt.getopt(sys.argv[1:], 'mhc:', ['help', 'command='])
-
-    if not args:
-        print(_usage)
+    import argparse
+
+    parser = argparse.ArgumentParser(prog="pdb",
+                                     description=_usage,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter,
+                                     allow_abbrev=False)
+
+    parser.add_argument('-c', '--command', action='append', default=[], metavar='command')
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('-m', metavar='module')
+    group.add_argument('pyfile', nargs='?')
```

Comment:
This is a slight work-around for limitations in argparse -- we want the mutually exclusive group behaviour, so ``nargs='?'`` is needed.

Reply:
Which limitation? Is there is a reason to use the wrong *nargs* on purpose, please add a comment to explain why.

Reply:
> Which limitation? Is there is a reason to use the wrong _nargs_ on purpose, please add a comment to explain why.

I don't think this is *wrong*. Consider it as - you can either pass a *module* with `-m` OR a *pyfile*. So *pyfile* is optional, you can pass no *pyfile*.

Reply:
That's not how exclusive group works. Either you pass a module or a pyfile. pyfile is not optional.

Reply:
> That's not how exclusive group works. Either you pass a module or a pyfile. pyfile is not optional.

I think either pass a module or a pyfile -> pyfile is optional, it's literally in a either/or sentence.

Mutually exclusive group requires all the options in the group to be "optional". What's your proposal for this? You can't use `nargs=1` because that would make `pyfile` a required argument - and it's not. It's absolutely fine if you don't pass a pyfile when you pass a module (actually you can't pass a pyfile when you pass a module).

Reply:
Sorry, I misunderstood how exclusive groups work. It's kind of surprising. Apparently, you must use ``nargs='?'``. It's just counter intuitive to me.

```py
import argparse
parser = argparse.ArgumentParser(prog='PROG')
group = parser.add_argument_group()
exclusive_group = group.add_mutually_exclusive_group(required=True)
exclusive_group.add_argument('-m', metavar='MODULE')
exclusive_group.add_argument('pyscript', nargs='?')
print(parser.parse_args(['-m', 'mymod']))
print(parser.parse_args(['myscript']))
print(parser.parse_args([]))
```

Output:

```
$ python3 x.py 
Namespace(m='mymod', pyscript=None)
Namespace(m=None, pyscript='myscript')
usage: PROG [-h] (-m MODULE | pyscript)
PROG: error: one of the arguments -m pyscript is required
```

Reply:
> It's just counter intuitive to me.

This bit of argparse's design is odd, I agree.

A